File: experiments/source/python/saxpy_curried.py
import loom


@loom.entry
def saxpy_curried(a: float, x: loom.Tensor1, y: loom.Tensor1) -> loom.Tensor1:
    def blend(a: float, xi: float, yi: float) -> float:
        return a * xi + yi

    return loom.Tensor1.map2(blend(a), x, y)

File: experiments/source/cpp/saxpy_curried.cpp
#include <loom/loom.hpp>

LOOM_ENTRY loom::Tensor1 saxpy_curried(float a, loom::Tensor1 x, loom::Tensor1 y) {
  auto blend = [](float a, float xi, float yi) -> float { return (a * xi) + yi; };
  return loom::Tensor1::map2(loom::partial(blend, a), x, y);
}
